src/main/java/io/appium/java_client/flutter/FlutterAndroidDriver.java

```diff
@@ -0,0 +1,134 @@
+import io.appium.java_client.flutter.commands.DragAndDropOptions;
+import io.appium.java_client.flutter.commands.LongPressOptions;
+import io.appium.java_client.flutter.commands.ScrollOptions;
+import io.appium.java_client.flutter.commands.WaitOptions;
+import io.appium.java_client.service.local.AppiumDriverLocalService;
+import io.appium.java_client.service.local.AppiumServiceBuilder;
+import org.openqa.selenium.Capabilities;
+import org.openqa.selenium.remote.HttpCommandExecutor;
+import org.openqa.selenium.remote.http.ClientConfig;
+import org.openqa.selenium.remote.http.HttpClient;
+
+import java.net.URL;
+
+/**
+ * Custom AndroidDriver implementation with additional Flutter-specific capabilities.
+ */
+public class FlutterAndroidDriver extends AndroidDriver implements FlutterDriver {
+
+    public FlutterAndroidDriver(HttpCommandExecutor executor, Capabilities capabilities) {
+        super(executor, capabilities);
+    }
+
+    public FlutterAndroidDriver(URL remoteAddress, Capabilities capabilities) {
+        super(remoteAddress, capabilities);
+    }
+
+    public FlutterAndroidDriver(URL remoteAddress, HttpClient.Factory httpClientFactory, Capabilities capabilities) {
+        super(remoteAddress, httpClientFactory, capabilities);
+    }
+
+    public FlutterAndroidDriver(AppiumDriverLocalService service, Capabilities capabilities) {
+        super(service, capabilities);
+    }
+
+    public FlutterAndroidDriver(
+            AppiumDriverLocalService service, HttpClient.Factory httpClientFactory, Capabilities capabilities) {
+        super(service, httpClientFactory, capabilities);
+    }
+
+    public FlutterAndroidDriver(AppiumServiceBuilder builder, Capabilities capabilities) {
+        super(builder, capabilities);
+    }
+
+    public FlutterAndroidDriver(
+            AppiumServiceBuilder builder, HttpClient.Factory httpClientFactory, Capabilities capabilities) {
+        super(builder, httpClientFactory, capabilities);
+    }
+
+    public FlutterAndroidDriver(HttpClient.Factory httpClientFactory, Capabilities capabilities) {
+        super(httpClientFactory, capabilities);
+    }
+
+    public FlutterAndroidDriver(ClientConfig clientConfig, Capabilities capabilities) {
+        super(clientConfig, capabilities);
+    }
+
+    public FlutterAndroidDriver(AppiumClientConfig appiumClientConfig, Capabilities capabilities) {
+        super(appiumClientConfig, capabilities);
+    }
+
+    public FlutterAndroidDriver(Capabilities capabilities) {
+        super(capabilities);
+    }
+
+    public FlutterAndroidDriver(URL remoteSessionAddress, String automationName) {
+        super(remoteSessionAddress, automationName);
+    }
+
+    /**
+     * Waits for an element to become visible on the Flutter application.
+     *
+     * @param option WaitOptions specifying the element and optional timeout.
+     */
+    @Override
+    public void waitForVisible(WaitOptions option) {
+        CommonFlutterCommands.waitForVisible(this, option);
```

Comment:
Agree on @mykola-mokhnach comments all these new driver part can be separate PR. And on your next PR it can be just FlutterCommands instead of CommonFlutterCommands.